### ISSUE TITLE
Lint against hyphens at the end of lines

### ISSUE THREAD

**Issue by JelleZijlstra** [sphinx-contrib/sphinx-lint#54]
PEP 520 has this text in its rst:

```
Additionally, this PEP requires that the default class definition
namespace be ordered (e.g. ``OrderedDict``) by default.  The long-
lived class namespace (``__dict__``) will remain a ``dict``.
```
This gets rendered as:

![Screen Shot 2023-02-24 at 4 35 53 PM](https://user-images.githubusercontent.com/906600/221325556-ee2c0759-674e-4c0e-abe8-c94b4aeb1fb6.png)

But there should be no space after the hyphen in the output.

A regex like `'[a-z]-$'` would catch this with few if any false positives. Could sphinx-lint add a rule looking for this issue?

**Comment by hugovk** [comment]
Please see PR https://github.com/sphinx-contrib/sphinx-lint/pull/56.
